proto/cosmos/bank/v1beta1/tx.proto

```diff
@@ -47,3 +53,37 @@ message MsgMultiSend {
 // MsgMultiSendResponse defines the Msg/MultiSend response type.
 message MsgMultiSendResponse {}
 
+// MsgSetSendEnabled is the Msg/SetSendEnabled request type.
+//
+// Only entries to add/update/delete need to be included.
+// Existing SendEnabled entries that are not included in this
+// message are left unchanged.
+message MsgSetSendEnabled {
+  option (cosmos.msg.v1.signer) = "authority";
+
+  string authority = 1 [(cosmos_proto.scalar) = "cosmos.AddressString"];
```

Comment:
what is an authority here?

Reply:
this shouldn't be a message executed by a normal address. It should be a proposal content

Reply:
I modeled this one after the `upgrade.v1beta1` package's `MsgSoftwareUpgrade` and `MsgCancelUpgrade` messages. As far as I know, those are the only examples of the new way to do the governance proposal messages.

The `GetSigners()` function returns just this `authority` field; it has to be the address of whatever has signed the message. In the `MsgServer`'s `SetSendEnabled` function, that address is required to be a specific value: `authtypes.NewModuleAddress(govtypes.ModuleName).String()` (or whatever else might be provided when creating the keeper). That ends up being just an address based on the module's name (hashed then truncated), and doesn't come from an associated private key.

The governance module is able to pass the message into the `SetSendEnabled` endpoint, though, because that part happens well after the point when `GetSigners()` is called and enforced. But if anyone tried to call the endpoint directly, they'd have to figure out the private key needed to sign for that address.

See [x/gov and x/group alignment #9066](https://github.com/cosmos/cosmos-sdk/discussions/9066) for more info.

Unit Tests that demonstrate this can be found in the `TestMsgSetSendEnabled` test in `x/bank/app_test.go` tests:
* [wrong authority](https://github.com/cosmos/cosmos-sdk/blob/0fbeb0c96e537856cf60c706d6bd69b0e7c62280/x/bank/app_test.go#L372-L386): The `MsgSetSendEnabled` is submitted directly in a TX; it fails because the `authority` is the only address that `addr1` can sign for, which is not the specific address the keeper checks for.
* [right authority wrong signer](https://github.com/cosmos/cosmos-sdk/blob/0fbeb0c96e537856cf60c706d6bd69b0e7c62280/x/bank/app_test.go#L387-L401): (just added recently), Again, sent directly in a TX. The `authority` is correct, but now the TX isn't signed correctly. This test was added after your comments.
* [submitted good as gov prop](https://github.com/cosmos/cosmos-sdk/blob/0fbeb0c96e537856cf60c706d6bd69b0e7c62280/x/bank/app_test.go#L402-L411): Submitted as a governance proposal signed by the user, but with the correct `authority` address.
* [submitted bad as gov prop](https://github.com/cosmos/cosmos-sdk/blob/0fbeb0c96e537856cf60c706d6bd69b0e7c62280/x/bank/app_test.go#L412-L424):  Part of a governance proposal, but with an invalid `MsgSetSendEnabled`. This just makes sure `ValidateBasic` is called when the message is part of a governance proposal.